### ISSUE TITLE
Override URL Helper schema to be HTTPS always

### ISSUE THREAD

**Issue by xperiandri** [OrchardCMS/OrchardCore#10328]
### Is your feature request related to a problem? Please describe.
If you put Orchard Core website under the reverse proxy where the proxy communicates with Orchard Core website via HTTP but serves it to the world by HTTPS all the links (especially SEO-related) become invalid (pointing to HTTP although page served from HTTPS).

### Describe the solution you'd like
An option that allows enabling the overriding scheme to be HTTPS always for any HTTP request.

### Describe alternatives you've considered
Communicate with HTTPS from nginx to Orchard Core website. But it adds unnecessary overhead.

**Comment by sebastienros** [comment]
The reverse proxy middleware is enabled by default in ASP.NET (and Orchard Core), however it will only be active if your settings are configured to do so, either in the appsettings.json or the environment variable.

c.f. 
https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-5.0#forward-the-scheme-for-linux-and-non-iis-reverse-proxies

Maybe we should add this in the default appsettings, commented, with an explanation on why to configure it.

**Comment by hyzx86** [comment]
Because OC needs to use host to distinguish tenants, if the tenant cannot be found, the reverse proxy function on the tenant cannot be applied

I plan to use domain name universal resolution to distinguish different tenants, but after I set the domain name for the default site, all requests will return blank...

**Comment by hyzx86** [comment]
> Maybe we should add this in the default appsettings, commented, with an explanation on why to configure it.

Yes, and it should only be enabled by the default site. The tenant should not manage the reverse proxy settings
It should be a function that only the site owner can manage

**Comment by hishamco** [comment]
I think we already have this in the `appsettings.json`
